### PR TITLE
Upgrade to Webpack 4

### PR DESCRIPTION
**Pull request by copperwall**
This updates the package.json to use Webpack 4 and updates the
webpack.config.js to be compatible with the new interface.

### CODE REVIEW

**Comment by djmetzle** [comment]
That's it? CR :question:

**Comment by copperwall** [comment]
Here's some comparisons between production and development builds between webpack 3 and 4:

## Webpack 4

### Production Mode:
```
Hash: f55a80b62a054cbc8cf1
Version: webpack 4.0.0
Time: 3605ms
Built at: 2/26/2018 11:31:52 AM
    Asset     Size  Chunks             Chunk Names
bundle.js  223 KiB       0  [emitted]  main
Entrypoint main = bundle.js
   [6] ./public/js/appearanceUtils.js 2.6 KiB {0} [built]
  [13] ./public/js/IndicatorFilter.js 2.89 KiB {0} [built]
  [15] ./views/standard/spec/utils.js 1.2 KiB {0} [built]
  [16] ./public/js/Templates.js 1.79 KiB {0} [built]
  [29] ./public/js/pullManager.js 1.68 KiB {0} [built]
  [43] ./public/js/ConnectionManager.js 1.05 KiB {0} [built]
  [44] ./public/js/pageIndicatorHandler.js 1.76 KiB {0} [built]
  [45] ./views/standard/spec/debugIndicators.js 2.75 KiB {0} [built]
  [50] ./views/standard/spec/pageIndicators.js 4.67 KiB {0} [built]
  [52] ./views/standard/spec/index.js 4.35 KiB {0} [built]
  [58] ./public/js/Column.js 6.28 KiB {0} [built]
  [59] ./public/js/ElementFilter.js 2.07 KiB {0} [built]
  [60] ./public/js/PullFilter.js 3.25 KiB {0} [built]
  [61] ./public/js/Pull.js 4.32 KiB {0} [built]
  [85] ./public/js/main.js 2.85 KiB {0} [built]
    + 71 hidden modules
```
### Development Mode:
```
Hash: bcca3fdfb631d8acface
Version: webpack 4.0.0
Time: 4554ms
Built at: 2/26/2018 12:08:41 PM
    Asset     Size  Chunks                    Chunk Names
bundle.js  703 KiB    main  [emitted]  [big]  main
Entrypoint main [big] = bundle.js
[./public/js/Column.js] 6.28 KiB {main} [built]
[./public/js/ConnectionManager.js] 1.05 KiB {main} [built]
[./public/js/ElementFilter.js] 2.07 KiB {main} [built]
[./public/js/IndicatorFilter.js] 2.89 KiB {main} [built]
[./public/js/Pull.js] 4.32 KiB {main} [built]
[./public/js/PullFilter.js] 3.25 KiB {main} [built]
[./public/js/Templates.js] 1.79 KiB {main} [built]
[./public/js/appearanceUtils.js] 2.6 KiB {main} [built]
[./public/js/main.js] 2.85 KiB {main} [built]
[./public/js/pageIndicatorHandler.js] 1.76 KiB {main} [built]
[./public/js/pullManager.js] 1.68 KiB {main} [built]
[./public/js/socket.js] 562 bytes {main} [built]
[./views/standard/spec/index.js] 4.35 KiB {main} [built]
[./views/standard/spec/pageIndicators.js] 4.67 KiB {main} [built]
[./views/standard/spec/utils.js] 1.2 KiB {main} [built]
    + 71 hidden modules
```
## Webpack 3

### Production Mode:
```
Hash: 918d516a6daa3d0b9c35
Version: webpack 3.8.1
Time: 6030ms
    Asset    Size  Chunks             Chunk Names
bundle.js  230 kB       0  [emitted]  main
   [6] ./public/js/appearanceUtils.js 2.66 kB {0} [built]
  [13] ./public/js/pullManager.js 1.72 kB {0} [built]
  [26] ./public/js/Templates.js 1.83 kB {0} [built]
  [29] ./public/js/IndicatorFilter.js 2.96 kB {0} [built]
  [30] ./public/js/main.js 2.92 kB {0} [built]
  [55] ./public/js/PullFilter.js 3.33 kB {0} [built]
  [56] ./public/js/ElementFilter.js 2.12 kB {0} [built]
  [57] ./public/js/Column.js 6.43 kB {0} [built]
  [63] ./views/standard/spec/index.js 4.46 kB {0} [built]
  [65] ./views/standard/spec/pageIndicators.js 4.78 kB {0} [built]
  [66] ./views/standard/spec/indicators.js 11 kB {0} [built]
  [69] ./views/standard/spec/columns.js 10.9 kB {0} [built]
  [70] ./views/standard/spec/debugIndicators.js 2.82 kB {0} [built]
  [71] ./public/js/pageIndicatorHandler.js 1.81 kB {0} [built]
  [72] ./public/js/ConnectionManager.js 1.08 kB {0} [built]
    + 71 hidden modules
```
### Development Mode:
```
Hash: 8a77ed7677758d12760f
Version: webpack 3.8.1
Time: 6623ms
    Asset     Size  Chunks                    Chunk Names
bundle.js  1.78 MB       0  [emitted]  [big]  main
   [6] ./public/js/appearanceUtils.js 2.66 kB {0} [built]
  [13] ./public/js/pullManager.js 1.72 kB {0} [built]
  [26] ./public/js/Templates.js 1.83 kB {0} [built]
  [29] ./public/js/IndicatorFilter.js 2.96 kB {0} [built]
  [30] ./public/js/main.js 2.92 kB {0} [built]
  [55] ./public/js/PullFilter.js 3.33 kB {0} [built]
  [56] ./public/js/ElementFilter.js 2.12 kB {0} [built]
  [57] ./public/js/Column.js 6.43 kB {0} [built]
  [63] ./views/standard/spec/index.js 4.46 kB {0} [built]
  [65] ./views/standard/spec/pageIndicators.js 4.78 kB {0} [built]
  [66] ./views/standard/spec/indicators.js 11 kB {0} [built]
  [69] ./views/standard/spec/columns.js 10.9 kB {0} [built]
  [70] ./views/standard/spec/debugIndicators.js 2.82 kB {0} [built]
  [71] ./public/js/pageIndicatorHandler.js 1.81 kB {0} [built]
  [72] ./public/js/ConnectionManager.js 1.08 kB {0} [built]
    + 71 hidden modules
```
## Takeaways

* Build time for both development and production builds is significantly decreased. Production: 6.6s vs 3.6s. Development 6.6s vs 4.5s
* Production builds are slightly smaller. 223kb vs 230kb
* Development builds are about half as large. We're using the default source map option now, so I'm not sure how much of a change that makes vs just webpack 3 to 4 changes.

File: webpack.config.js [hunk 1]
```diff
@@ -34,13 +34,12 @@ module.exports = {
       new webpack.ProvidePlugin({
         $: "jquery",
         jQuery: "jquery"
-      }),
-      new webpack.optimize.UglifyJsPlugin()
```
Comment (copperwall) [inline webpack.config.js]: This is builtin now :).

**Comment by n-fisher** [comment]
CR 👍

**Pull request merged**